### ISSUE TITLE
Failed to build due to hGetContents 

### ISSUE THREAD

**Issue by gshen42** [agda/agda#5739]
I'm building Agda 2.6.2.1 on macOS with `ghc` 9.2.1 and `cabal` 3.6.2.0. The build mostly succeeded, but at the very end when it tried to type check some primitive module, it threw an error which says `hGetContents: invalid argument (invalid byte sequence)`. Here is the relevant output:
```
[400 of 400] Compiling Agda.Main        ( src/full/Agda/Main.hs, dist/build/Agda/Main.o, dist/build/Agda/Main.dyn_o )
Preprocessing executable 'agda' for Agda-2.6.2.1..
Building executable 'agda' for Agda-2.6.2.1..
[1 of 1] Compiling Main             ( src/main/Main.hs, dist/build/agda/agda-tmp/Main.o )
Linking dist/build/agda/agda ...
Installing executable agda-mode in /Users/gan/.cabal/store/ghc-9.2.1/incoming/new-28765/Users/gan/.cabal/store/ghc-9.2.1/Agd-2.6.2.1-4bd94664/bin
Warning: The directory
/Users/gan/.cabal/store/ghc-9.2.1/incoming/new-28765/Users/gan/.cabal/store/ghc-9.2.1/Agd-2.6.2.1-4bd94664/bin
is not in the system search path.
Installing library in /Users/gan/.cabal/store/ghc-9.2.1/incoming/new-28765/Users/gan/.cabal/store/ghc-9.2.1/Agd-2.6.2.1-4bd94664/lib
Installing executable agda in /Users/gan/.cabal/store/ghc-9.2.1/incoming/new-28765/Users/gan/.cabal/store/ghc-9.2.1/Agd-2.6.2.1-4bd94664/bin
Warning: The directory
/Users/gan/.cabal/store/ghc-9.2.1/incoming/new-28765/Users/gan/.cabal/store/ghc-9.2.1/Agd-2.6.2.1-4bd94664/bin
is not in the system search path.
/private/var/folders/vn/qsn8r0ps59x6xr27xkyh_vvc0000gn/T/cabal-install.-28765/dist-newstyle/tmp/src-28765/Agda-2.6.2.1/src/data/lib/prim/Agda/Primitive.agda:
hGetContents: invalid argument (invalid byte sequence)
Error: Failed to typecheck /private/var/folders/vn/qsn8r0ps59x6xr27xkyh_vvc0000gn/T/cabal-install.-28765/dist-newstyle/tmp/src-28765/Agda-2.6.2.1/src/data/lib/prim/Agda/Builtin/Bool.agda!
Generating Agda library interface files...
... /private/var/folders/vn/qsn8r0ps59x6xr27xkyh_vvc0000gn/T/cabal-install.-28765/dist-newstyle/tmp/src-28765/Agda-2.6.2.1/src/data/lib/prim/Agda/Builtin/Bool.agda
cabal: Failed to build Agda-2.6.2.1. See the build log above for details.
```

**Comment by andreasabel** [comment]
@gshen42 Could this be a _locale_ issue?  Please read up on #2922 and similar issues (google!).

**Comment by nad** [comment]
The error message mentions `hGetContents`, and Agda should not make use of the locale to decide how to read something. Unfortunately that is not always the case (#5741, #5742).

It would perhaps be easier to debug this problem if we knew which instance of `hGetContents` causes it. One option is to compile Agda with profiling turned on, and use `+RTS -xc` to get a stack trace. I guess the error message comes from `Setup.hs`, so one could add `"+RTS", "-xc"` to the end of the following list:

https://github.com/agda/agda/blob/9399b3a05e56b9957265949d2eeaba64e004e799/Setup.hs#L110-L114

However, we could also fix #5741 and #5742 and see if this addresses the problem.

**Comment by gshen42** [comment]
Sorry, I didn't mention I was doing this on a M1 Mac, after doing some research, it seems like GHC 9.2.1 has some issues dealing with unicode (see https://gitlab.haskell.org/ghc/ghc/-/issues/20677 and https://discourse.haskell.org/t/psa-9-2-1-aarch64-miscompilation/3638), so it's not Agda's fault.

I reverted back to GHC 9.0.2 and everything works fine now.

**Comment by andreasabel** [comment]
Let's keep it open until it is fixed (either by us or upstream).

**Comment by nad** [comment]
I've at least fixed #5741 and #5742.

**Comment by andreasabel** [comment]
I find the GHC 9.2.2 release notes somewhat inconclusive: https://downloads.haskell.org/ghc/9.2.2/docs/html/users_guide/9.2.2-notes.html
@gshen42 : Is the build problem fixed with GHC 9.2.2?

**Comment by gshen42** [comment]
Just tested it, the build succeeded with GHC 9.2.2, thanks for the help!
